specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingSubscription.json

```diff
@@ -400,244 +749,605 @@
+          "readOnly": true,
           "x-ms-enum": {
-            "modelAsString": true,
-            "name": "BillingSubscriptionStatusType"
+            "name": "BillingSubscriptionStatus",
+            "modelAsString": true
           }
         },
         "subscriptionId": {
           "description": "The ID of the subscription.",
-          "format": "uuid",
-          "readOnly": true,
-          "type": "string"
+          "type": "string",
+          "readOnly": true
         },
         "suspensionReasons": {
-          "description": "The suspension reason for a subscription. Applies only to subscriptions in Microsoft Online Services Program billing accounts.",
+          "description": "The suspension reason for a subscription. This field is not available for Enterprise Agreement billing accounts.",
+          "type": "array",
           "items": {
-            "type": "string"
+            "enum": [
+              "None",
+              "Cancelled",
+              "PastDue",
+              "SuspiciousActivity",
+              "Other",
+              "Transferred",
+              "PolicyViolation",
+              "SpendingLimitReached",
+              "Expired"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "SubscriptionStatusReason",
+              "modelAsString": true
+            }
           },
           "readOnly": true,
-          "type": "array"
+          "x-ms-identifiers": []
+        },
+        "suspensionReasonDetails": {
+          "description": "The suspension details for a subscription. This field is not available for Enterprise Agreement billing accounts.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BillingSubscriptionStatusDetails"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
         }
       }
     },
-    "BillingSubscriptionsListResult": {
-      "description": "The list of billing subscriptions.",
+    "BillingSubscriptionStatusDetails": {
+      "description": "The suspension details for a subscription. This field is not available for Enterprise Agreement billing accounts.",
+      "type": "object",
       "properties": {
-        "nextLink": {
-          "description": "The link (url) to the next page of results.",
-          "readOnly": true,
-          "type": "string"
-        },
-        "totalCount": {
-          "description": "Total number of records.",
-          "format": "int32",
-          "readOnly": true,
-          "type": "number"
+        "effectiveDate": {
+          "description": "The suspension effective date for a subscription. This field is not available for Enterprise Agreement billing accounts.",
+          "type": "string",
+          "readOnly": true
         },
-        "value": {
-          "description": "The list of billing subscriptions.",
-          "items": {
-            "$ref": "#/definitions/BillingSubscription"
-          },
+        "reason": {
```

Comment:
Why do we have both this and BillingSubscription.suspensionReasons?

Reply:
I believe that `SuspensionReason` is for legacy, but @ramkd12  and @amberbhargava  to comment.